src/renderer/screens/manager/Dashboard.js

```diff
@@ -10,15 +11,32 @@ import TrackPage from "~/renderer/analytics/TrackPage";
 import Box from "~/renderer/components/Box";
 import { command } from "~/renderer/commands";
 import FirmwareUpdate from "./FirmwareUpdate";
+import { getCurrentDevice } from "~/renderer/reducers/devices";
 
 type Props = {
   device: Device,
   deviceInfo: DeviceInfo,
   result: ?ListAppsResult,
+  onReset?: () => void,
 };
 
-const Dashboard = ({ device, deviceInfo, result }: Props) => {
+const Dashboard = ({ device, deviceInfo, result, onReset }: Props) => {
+  /** locks and unlocks the device connection state reset ie: firmware update */
+  const [resetLock, setResetLock] = useState(false);
+  const lockReset = useCallback(() => setResetLock(true), [setResetLock]);
+  const unlockReset = useCallback(() => setResetLock(false), [setResetLock]);
+  /** we listen to the redux state to detect the current device connected */
+  const currentDevice = useSelector(getCurrentDevice);
```

Comment:
as this component seems to be connect, I would suggest instead to just have:

```
resetLock = useSelect(aSelectorThatCheckIfFirmwareUpdateModalIsOpened)
```

that would simplify some code below.